### DISCUSSION TITLE
How to make sub queries

### DISCUSSION THREAD

**Discussion by Abenezerdus** [supabase/supabase#12442] [Questions] [answered]
Hi
Is there a way to convert this sql query

 select * from recipe where state = 'published' and id in (
    select id from recipe where calories > 250 and calories < 260 or calories > 950 and calories < 958
 );

into javascript client library query?

**Comment by GaryAustin1** [comment, marked as answer]
Not quite sure what you query is doing.
Looks like your "sub query" is on the same table as your query (recipe).  If so, then you could just chain all the filters (published, gt 250, lt 260, etc) on recipe and don't need the in.

The client does have the ability using !inner after an embedded table and then filters with embedded_table_name.col in the filters to do filtering on the embedded table.   But you don't show a join/embedded table.

**Reply by Abenezerdus** [reply]
The query is doing multiple check filters 

For example if user selects these ranges
[x] 100 - 400 calories
[] 400 - 800 calories
[x] 800 - 1200 calories

i can translate it to subquery lik
select * from recipe where state = 'published' and id in (
  select id from recipe where calories > 100 and calories < 400 or calories > 800 and calories < 1200
);

but if i use 
query
      .gt('calories', 100)
      .lt('calories', 400)
      .gt('calories', 800)
      .lt('calories', 1200);

i don't get results for the 800 to 1200 range

**Reply by GaryAustin1** [reply]
Yes you want to use .or filter with and between the pairs.
https://supabase.com/docs/reference/javascript/or

![image](https://user-images.githubusercontent.com/54564956/219077017-ad381334-3757-4166-83cf-6412cc5ce201.png)

**Reply by Abenezerdus** [reply]
Ok thanks, this solved it
